microprot/snakemake/Snakefile

```diff
@@ -3,192 +3,276 @@
+
 
 rule split_Pfam:
     input:
-        'results/{seq}/06-search_Pfam/{seq}'
+        rules.search_Pfam.output[0]
     output:
-        'results/{seq}/07-split_Pfam/{seq}.split_pfam'
+        config['MICROPROT_MAIN']+'results/06-split_Pfam/{seq}/{seq}'
     log:
-        'results/log/{seq}.split_pfam.log'
+        'results/log/{seq}/split_pfam.log'
     run:
-        for inp in open(input[0], 'r').readlines():
-            mask_sequence(inp, output[0], config['split_Pfam']['params'])
+        split_x(input[0], output[0],
+                step="Pfam",
+                frag_len=config['split_Pfam']['params']['min_fragment_length'],
+                e_val=config['split_Pfam']['params']['max_evalue'],
+                prob=config['split_Pfam']['params']['min_prob'], log=log[0])
+
 
 rule MSA_hhblits:
     input:
-        split='results/{seq}/07-split_Pfam/{seq}.split_pfam',
-        seq='data/{seq}.fasta'
+        config['MICROPROT_MAIN']+'results/06-split_Pfam/{seq}/{seq}'
     output:
-        'results/{seq}/08-MSA_hhblits/{seq}'
+        config['MICROPROT_MAIN']+'results/07-MSA_hhblits/{seq}/{seq}'
     log:
-        'results/log/{seq}.msa_hhblits.log'
+        'results/log/{seq}/msa_hhblits.log'
     run:
-        shell_f = shell('{paths[TOOLS][hhsuite]}/bin/hhblits -i {domain} \
-                         {config[MSA_hhblits][params]} \
-                         -d {paths[DBS][hhblits_db]} \
-                         -o {out} \
-                         -oa3m {a3m}')
-        search_X(input['seq'], input['split'], output[0], shell_f)
+        """
+        it's essentially a copy of `search_x` function, but here we're using
+        HHblits and it has a slightly different commandline input construction
+        """
+        indir = snakemake_helpers.trim(input[0], '/')
+        outdir = snakemake_helpers.trim(output[0], '/')
+        # here, we process both `match` and `non_match` sequences
+        # Pfam is our "domain splitting" method
+        for _match in glob('%s/%s' % (indir, '*match')):
+            if snakemake_helpers.not_empty(_match):
+                process_fasta.split_fasta(process_fasta.extract_sequences(
+                                          _match), outdir=outdir)
+        for hh_inp in glob('%s/%s' % (outdir, '*.fasta')):
+            out_root = snakemake_helpers.trim(hh_inp, '.')
+            shell('{paths[TOOLS][hhsuite]}/hhblits -i {hh_inp} \
+                   {config[MSA_hhblits][params]} \
+                   -d {paths[DBS][hhblits_db]} \
+                   -o {out_root}.out \
+                   -oa3m {out_root}.a3m 2> {log}')
+        shell('touch {output}')
+
 
 rule MSA_ripe:
     input:
-        'results/{seq}/08-MSA_hhblits/{seq}'
+        config['MICROPROT_MAIN']+'results/07-MSA_hhblits/{seq}/{seq}'
     output:
-        not_ripe='results/{seq}/09-not_ripe/{seq}.msa',
-        ripe='results/{seq}/10-Rosetta/{seq}.msa'
+        config['MICROPROT_MAIN']+'results/log/{seq}/msa_ripe.log'
```

Comment:
Rule's output (`msa_ripe.log`) is written regardless of what happens downstream, so we should be good here. However, it is very likely that if there is no input for one of the `search_X` rules (e.g. if the entire sequence matches PDB or CM) we will get an error. That needs further investigation.
Comments in lines 94-98 outline how to potentially deal with this problem. What do you think?